types/invoker.go

```diff
@@ -26,32 +34,34 @@ func (i *Invoker) Invoke(topicMap *TopicMap, topic string, message *[]byte) {
 			gwURL := fmt.Sprintf("%s/function/%s", i.GatewayURL, matchedFunction)
 			reader := bytes.NewReader(*message)
 
-			body, statusCode, doErr := invokefunction(i.Client, gwURL, reader)
+			err, statusCode, headers, body := performInvocation(i.Client, gwURL, reader)
 
-			if doErr != nil {
-				log.Printf("Unable to invoke from %s, error: %s\n", matchedFunction, doErr)
+			if err != nil {
+				result[matchedFunction] = &InvocationResult{
+					StatusCode: statusCode,
+					Body:       nil,
+					Error:      err,
+				}
 				return
 			}
 
-			printBody := false
-			stringOutput := ""
-
 			if body != nil && i.PrintResponse {
-				stringOutput = string(*body)
-				printBody = true
+				stringOutput := string(*body)
+				log.Printf("Headers: %s", headers)
+				log.Printf("Response: [%d] from %s %s", statusCode, matchedFunction, stringOutput)
 			}
 
-			if printBody {
-				log.Printf("Response [%d] from %s %s", statusCode, matchedFunction, stringOutput)
-
-			} else {
-				log.Printf("Response [%d] from %s", statusCode, matchedFunction)
+			result[matchedFunction] = &InvocationResult{
+				StatusCode: statusCode,
+				Body:       body,
+				Error:      nil,
 			}
 		}
 	}
+	return result
 }
 
-func invokefunction(c *http.Client, gwURL string, reader io.Reader) (*[]byte, int, error) {
+func performInvocation(c *http.Client, gwURL string, reader io.Reader) (err error, statusCode int, responseHeaders http.Header, responseBody *[]byte) {
```

Comment:
Would it make sense to also return a struct (like `InvokationResult` above) instead of four individual response items?